File: practicas/practica1_rb/WeaponType.rb
#encoding: utf-8

#Enumerado que representa a los tipos de armas del juego.
#Cada valor posible del tipo de enumerado tendrá un valor numérico concreto
#igual a la potencia de disparo de cada tipo de armas

module Deepspace
  module WeaponType

    #Clase auxiliar para representar el enumerado
    class Type

      #Constructor con parametros de la clase Type
      def initialize(pwr)
        @power = pwr
      end

      #Metodo consultor del atributo power
      def power
        @power
      end
    end

    LASER = Type.new 2.0
    MISSILE = Type.new 3.0
    PLASMA = Type.new 4.0

  end
end

File: practicas/practica4/src/main/java/deepspace/WeaponType.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *  representa a los tipos de armas del juego y no será simplemente una lista devalores, sino que tendrá cierta funcionalidad adicional.
 * En este caso, cada valor posible del tipoenumerado tendrá asociado un valor numérico concreto igual a la potencia
 * de disparo de cada tipode arma.
 */
public enum WeaponType {
    LASER(2.0f), MISSILE(3.0f), PLASMA(4.0f);
    
    private float power;

    WeaponType(float power) {
        this.power = power;
    }
    
    float getPower() {
        return this.power;
    }
}
